stdlib/list.gr

```diff
@@ -195,3 +195,25 @@ let unique = (list) => {
   }
   iter(list, [])
 }
+
+let rec drop = (n, list) => {
+  if (n < 0) {
+    fail 'number of items to drop cannot be a negative number'
+  } else {
+    match ((n, list)) {
+      | (_,[]) => []
+      | (n, _) when n == 0 => list
+      | (n, [first, ...rest]) => drop(n - 1, rest)
+    }
+  }
+}
+
+let rec dropWhile = (fn, list) => {
+  match (list) {
+    | [] => list
+    | [first, ...rest] => if (fn(first)) dropWhile(fn, rest) else list
+  }
+}
```

Comment:
I'm not sure this one is tail-recursive but from what I saw, others from this file like `count` are not. I don't know at which point it matters?

Reply:
Hmm, I think we want to move towards making both this and `count` tail-recursive. @ospencer should we make a separate issue for this, or do we want to include making this/these tail-recursive inside of this PR?

Reply:
Maybe it's really early for me, but is this not tail-recursive? The call to `dropWhile` is in the tail position of the true branch of a tail position `if`.

`count` does need to be changed though. Made https://github.com/grain-lang/grain/issues/319.

Reply:
Oh, yes, lol. I missed that; you're right.

Reply:
Thanks for the review, I was indeed not sure for the `dropWhile` :)